api/base/filters.py

```diff
@@ -105,11 +105,12 @@ def _get_field_or_error(self, field_name):
 
         :raises InvalidFilterError: If the filter field is not valid
         """
-        if field_name not in self.serializer_class._declared_fields:
+        serializer_class = self.serializer_class
```

Comment:
Prefer `.get_serializer_class()` -- `filterable_fields` can differ between requests, and the assigned `.serializer_class` isn't the same as the return value of the method